src/Topic0/AbsFactory.java

```diff
@@ -0,0 +1,35 @@
+package Topic0;
+
+import java.sql.*;
+import java.lang.*;
+import java.lang.String;
+import java.lang.Integer;
+
+//Ex 2
+
+public abstract class AbsFactory {
+
+  String type = "tipo";
+  String driver = "com.mysql.jdbc.Driver";
+  String url = "jdbc:mysql://";
+  String pass = "contraseña";
+  String usr = "usuario";
+
+  //Connect according Db Type
+  public String getConection(String type, String driver, String url, String pass, String usr) {
+    String option = type;
+    if (option == "Oracle") {
```

Comment:
Here you should be able to do something like

```suggestion
   swtich(type) {
       case "Oracle": return new Oracle(driver, url, pass, usr);
       case "Postgre": return new Postgre(driver, url, pass, usr);
       ...
       default:
           throw new IllegalArgumentException("Unable to recognice " + type);
   }
```